tensorbay/client/tests/test_gas.py

```diff
@@ -0,0 +1,346 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+
+import os
+
+import pytest
+
+from ...dataset import Dataset
+from ...exception import DatasetTypeError, ResourceNotExistError
+from .. import gas
+from ..dataset import DatasetClient, FusionDatasetClient
+from ..gas import GAS
+from ..status import Status
+from .utility import mock_response
+
+
+class TestGAS:
+    gas_client = GAS("Accesskey-********************************")
+
+    def test_generate_auth_storage_configs(self, mocker):
+        params = {"name": "cloud_config", "offset": 0, "limit": 128}
+        response_data = {
+            "configs": [
+                {
+                    "name": "cloud_config",
+                    "type": "azure",
+                    "accountName": "tensorbay",
+                    "containerName": "tensorbay",
+                },
+            ],
+            "offset": 0,
+            "recordSize": 1,
+            "totalCount": 1,
+        }
+        open_api_do = mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data=response_data),
+        )
+        assert (
+            next(self.gas_client._generate_auth_storage_configs("cloud_config"))
+            == response_data["configs"][0]
+        )
+
+        open_api_do.assert_called_once_with("GET", "auth-storage-configs", "", params=params)
+
+    @pytest.mark.parametrize("is_fusion", [True, False])
+    def test_get_dataset_with_any_type(self, mocker, is_fusion):
+        response_data = {
+            "name": "test",
+            "type": int(is_fusion),
+            "defaultBranch": "main",
+            "commitId": "4",
+            "updateTime": 1622693494,
+            "owner": "",
+            "id": "123456",
+        }
+        get_dataset = mocker.patch(
+            f"{gas.__name__}.GAS._get_dataset",
+            return_value=response_data,
+        )
+        dataset_client = self.gas_client._get_dataset_with_any_type("test")
+        dataset_type = FusionDatasetClient if is_fusion else DatasetClient
+        assert isinstance(dataset_client, dataset_type)
+        assert dataset_client._name == "test"
+        assert dataset_client._dataset_id == "123456"
+        assert dataset_client._status.commit_id == "4"
+        assert dataset_client._status.branch_name == "main"
+        get_dataset.assert_called_once_with("test")
+
+    def test_get_dataset_(self, mocker):
```

Comment:
check famous third party project and try to find out how to handle the test case of private function.